### ISSUE TITLE
Test time of template mode increased and not all relevant rules are tested

### ISSUE THREAD

**Issue by yuumasato** [ComplianceAsCode/content#8886]
#### Description of problem:
- After #8751 the Test Suite's template mode takes much longer to run and prints a lot of ERROR messages. :cry: 
- And not all rules using the template are tested

#### Steps to Reproduce:

1. Build your favorite content
2. Run the Test Suite on template mode, for example with `dconf_ini_file`, which is used by 2 rules only.
`$ time python3 tests/test_suite.py template  --libvirt qemu:///session rhel86 --datastream build/ssg-rhel8-ds.xml dconf_ini_file`

#### Actual Results:
It takes a lot of time to identify the rules it should test.
```
time python3 tests/test_suite.py template  --libvirt qemu:///session rhel86 --datastream build/ssg-rhel8-ds.xml dconf_ini_file
Setting console output to log level INFO                                                                                                                                                                                                                                                                                      
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/wsato/git/content/logs/template-custom-2022-06-03-1230/test_suite.log
ERROR - Rule 'xccdf_org.ssgproject.content_rule_service_apport_disabled' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-k8sb635r'
ERROR - Rule 'xccdf_org.ssgproject.content_rule_apt_conf_disallow_unauthenticated' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-k8sb635r'
ERROR - Rule 'xccdf_org.ssgproject.content_rule_apt_sources_list_official' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-k8sb635r'
ERROR - Rule 'xccdf_org.ssgproject.content_rule_service_cgconfig_disabled' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-k8sb635r'
ERROR - Rule 'xccdf_org.ssgproject.content_rule_service_cgred_disabled' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-k8sb635r'
ERROR - Rule 'xccdf_org.ssgproject.content_rule_service_irqbalance_enabled' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-k8sb635r'
...
ERROR - Rule 'xccdf_org.ssgproject.content_rule_kernel_config_legacy_vsyscall_xonly' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-k8sb635r'
INFO - xccdf_org.ssgproject.content_rule_dconf_gnome_lock_screen_on_smartcard_removal
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
INFO - Script missing.fail.sh using profile (all) OK

real    22m14.059s
user    18m37.027s
sys     0m16.574s

```

#### Expected Results:
More sane run time. 
On 4ce3626d28 I get this:
```
$ time python3 tests/test_suite.py template  --libvirt qemu:///session rhel86 --datastream build/ssg-rhel8-ds.xml dconf_ini_file
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/wsato/git/parallel-content/logs/template-custom-2022-06-03-1253/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_dconf_gnome_lock_screen_on_smartcard_removal
INFO - Script missing.fail.sh using profile (all) OK
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
INFO - xccdf_org.ssgproject.content_rule_dconf_gnome_disable_user_admin
INFO - Script missing.fail.sh using profile (all) OK
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK

real	6m2.663s
user	0m49.041s
sys	0m3.711s
```

#### Additional Information/Debugging Steps:

**Comment by yuumasato** [comment]
@matejak @jan-cerny Hi, would you be so kind and check what is going on?

I checked the code but there isn't anything obvious popping out to me, there doesn't seem to be any new time consuming function. The order of in which they are called changed, and that may be it....

**Comment by yuumasato** [comment]
Another issue I noticed now is that only one of the rules got tested: `dconf_gnome_lock_screen_on_smartcard_removal`

**Comment by jan-cerny** [comment]
I also think that it's caused by the different order of calls. While the order of operations might be great for the rule mode, it's not so good in the template mode.

What I found that there is a very inefficient way of checking if a rule is present in the given data stream, it always loads the data stream from file, which is a problem if we iterate over large amount of rules.

Moreover, now it produces an error message that a rule isn't present in the given benchmark, which is a good error message in the rule mode, where the user specifies the rules to be tested, but, in the template mode user specifies the template, so in the template mode it should just silently skip the rules that are present there.

I will look into this.
